pkg/api/status_api_handlers.go

```diff
@@ -231,3 +257,133 @@ func convertToPbGetCephOsdDumpResponse(osdDump types.CephOsdDumpResponse) *pb.Ge
 		StretchMode:      osdDump.StretchMode,
 	}
 }
+
+func convertToPbGetCephPgDumpReponse(pgDump *types.PgDumpResponse) (*pb.GetCephPgDumpResponse, error) {
+	// If the entire pgDump is nil, return an empty response
+	if pgDump == nil {
+		return &pb.GetCephPgDumpResponse{}, fmt.Errorf("pgDump is nil")
+	}
+
+	// Check if the PgMap is nil
+	if pgDump.PgMap == nil {
+		return &pb.GetCephPgDumpResponse{}, fmt.Errorf("pgDump.PgMap is nil")
+	}
+
+	pgStatsDelta := pb.PGStatsDelta{
+		StatSum:       pgDump.PgMap.PgStatsDelta.StatSum,
+		StoreStats:    pgDump.PgMap.PgStatsDelta.StoreStats,
+		LogSize:       pgDump.PgMap.PgStatsDelta.LogSize,
+		OndiskLogSize: pgDump.PgMap.PgStatsDelta.OndiskLogSize,
+		Up:            pgDump.PgMap.PgStatsDelta.Up,
+		Acting:        pgDump.PgMap.PgStatsDelta.Acting,
+		NumStoreStats: pgDump.PgMap.PgStatsDelta.NumStoreStats,
+		StampDelta:    pgDump.PgMap.PgStatsDelta.StampDelta.Timestamp,
+	}
+	var networkingPingTimes []*pb.OSDStatsSum_NetworkPingTime
+	for _, pingTime := range pgDump.PgMap.OsdStatsSum.NetworkPingTimes {
+		networkingPingTimes = append(networkingPingTimes, &pb.OSDStatsSum_NetworkPingTime{
+			Osd:        pingTime.Osd,
+			LastUpdate: pingTime.LastUpdate.Timestamp,
+			Interfaces: pingTime.Interfaces,
+		})
+	}
+	osdStatsSum := pb.OSDStatsSum{
+		UpFrom:             pgDump.PgMap.OsdStatsSum.UpFrom,
+		Seq:                pgDump.PgMap.OsdStatsSum.Seq,
+		NumPgs:             pgDump.PgMap.OsdStatsSum.NumPgs,
+		NumOsds:            pgDump.PgMap.OsdStatsSum.NumOsds,
+		NumPerPoolOsds:     pgDump.PgMap.OsdStatsSum.NumPerPoolOsds,
+		NumPerPoolOmapOsds: pgDump.PgMap.OsdStatsSum.NumPerPoolOmapOsds,
+		Kb:                 pgDump.PgMap.OsdStatsSum.Kb,
+		KbUsed:             pgDump.PgMap.OsdStatsSum.KbUsed,
+		KbUsedData:         pgDump.PgMap.OsdStatsSum.KbUsedData,
+		KbUsedOmap:         pgDump.PgMap.OsdStatsSum.KbUsedOmap,
+		KbUsedMeta:         pgDump.PgMap.OsdStatsSum.KbUsedMeta,
+		KbAvail:            pgDump.PgMap.OsdStatsSum.KbAvail,
+		Statfs:             pgDump.PgMap.OsdStatsSum.Statfs,
+		HbPeers:            pgDump.PgMap.OsdStatsSum.HbPeers,
+		SnapTrimQueueLen:   pgDump.PgMap.OsdStatsSum.SnapTrimQueueLen,
+		NumSnapTrimming:    pgDump.PgMap.OsdStatsSum.NumSnapTrimming,
+		NumShardsRepaired:  pgDump.PgMap.OsdStatsSum.NumShardsRepaired,
+		OpQueueAgeHist:     pgDump.PgMap.OsdStatsSum.OpQueueAgeHist,
+		PerfStat:           pgDump.PgMap.OsdStatsSum.PerfStat,
+		Alerts:             pgDump.PgMap.OsdStatsSum.Alerts,
+		NetworkPingTimes:   networkingPingTimes,
+	}
+
+	var pgStats []*pb.PGStat
```

Comment:
consider preallocating for slices of known size:

```suggestion
	pgStats:=make([]*pb.PGStat,len(pgDump.PgMap.PgStats))
```